transactions-v1/sell_item.cdc

```diff
@@ -20,10 +20,23 @@ transaction(saleItemID: UInt64, saleItemPrice: UFix64) {
         self.exampleTokenReceiver = acct.capabilities.get<&{FungibleToken.Receiver}>(/public/exampleTokenReceiver)
         assert(self.exampleTokenReceiver.check(), message: "Missing or mis-typed ExampleToken Receiver")
 
-        self.exampleNFTProvider = acct.capabilities.storage.issue<auth(NonFungibleToken.Withdraw) &{NonFungibleToken.Collection}>(
+        // The storage path for storing a withdraw capability to the nft collection
+        // Should preferrably be defined in the nft contract
+        let nftProviderCapPath = /storage/ExampleNFTProviderCap
+        var nftProviderCap = acct.storage.copy<Capability<auth(NonFungibleToken.Withdraw) &{NonFungibleToken.Collection}>>(from: nftProviderCapPath)
+        if nftProviderCap == nil || !nftProviderCap!.check() {
+            self.exampleNFTProvider = acct.capabilities.storage.issue<auth(NonFungibleToken.Withdraw) &{NonFungibleToken.Collection}>(
                 collectionData.storagePath
             )
-        assert(self.exampleNFTProvider.check(), message: "Missing or mis-typed ExampleNFT provider")
+            assert(self.exampleNFTProvider.check(), message: "Missing or mis-typed ExampleNFT provider")
+             // save capability to storage
+            acct.storage.save(
```

Comment:
the alternative to this is creating a tag for the capability, but the only way to access it is by looping through all capabilities which does not scale

Reply:
It's possible to get the `StorageCapabilityController` for Capabilities targeting a given storage path. Something like [this pattern](https://run.dnz.dev/snippet/c373ba103400a273). Of course, issuing & saving the cap to later copy works, though not sure that's a pattern we want to encourage. 

@joshuahannan I see you've approved, but wondering if you have thoughts on a preference here

Reply:
while i prefer the sample you shared, a lot of accounts already have thousands of capabilities issued for the same storage path and trying to fetch them all will most likely go past the computation limit, for example this account `0xead892083b3e2c6c`

Reply:
Oh wow, thanks for sharing! Agreed, this avoids that issue.

Reply:
tho, i tried the script with that account and it works even tho that storage path has 100s of thousands of capabilities https://run.dnz.dev/snippet/c452d63c9ea964b0

Reply:
if you think it wouldn't cause a computation limit error, i'm happy to make the change

Reply:
i think it would be helpful if the ci doesn't need an approval to run